rails_programming/assets_and_navigation/turbo_drive.md

```diff
@@ -0,0 +1,153 @@
+### Introduction
+
+Rails has always promoted itself as a framework that makes building a web application easy, but in a world built around the demands of users expecting lightning fast responses as well as increasingly complex UI it wasn't easy for Rails to offer a compelling vision for how they saw a way through that complexity while sticking to core Rails philosophies. However just before the release of Rails 7 we finally got a glimpse of how the Rails team saw the future of meeting these demands. Hotwire!
+
+Hotwire is actually an umbrella term for three different frameworks. These frameworks are:
+
+1.  Turbo
+2.  Stimulus
+3.  Strada
+
+Stimulus is something we'll cover later and you don't need to worry about Strada for the scope of this course, just be familiar with the name as you will see it mentioned from time-to-time.
+
+Turbo itself is also an umbrella term for several different techniques for creating fast and modern web applications. The technique we're going to focus on here is the one whose behaviour is part of Rails' core behaviour, Turbo Drive. The other techniques we'll cover in the Advanced Turbo lesson later.
+
+Links and Forms are the bedrock of most web applications. You need links to navigate around your site and forms to allow users to submit information to you. Turbo Drive is how Rails handles these in your applications.
+
+If a lot of what follows seems confusing, don't worry! A lot of Turbo Drive happens under the hood and therefore it means we often need to discuss the concept rather than the implementation. This can make it harder initially to grok, but luckily it does also mean you can often just go about your business and never have to actively think about it.
+
+### Learning Outcomes
+
+*Look through these now and then use them to test yourself after doing the assignment*
+
+* What is Turbolinks?
+* What is the purpose of Turbo Drive?
+* How does Turbo Drive handle page navigation?
+* What are the two kinds of page visit types in Turbo?
+* How can you disable Turbo Drive?
+
+### Turbolinks
+
+Before we can talk about Turbo, we should briefly mention its predecessor, Turbolinks. Turbolinks is **no longer in active development**! Despite that, it's still worth mentioning as it has been around for a while (its first major release was 2013) and it's possible that you will encounter existing applications or online articles that mention Turbolinks. The goal of Turbolinks was to capture the request for any clicked links in your web application and, if that link was to another page in your web app and not to an outside resource, then Turbolinks would keep the current page instance alive and only swap out the content between the `<body>` tags of the document. Because it didn't do a full page redirect it was faster because it didn't have to attempt to download any resources in the head part of the document when they were unlikely to have unchanged.
```

Comment:
I think it may be a little confusing to start with turbolinks. Imo this would fit better at the end as an aside. 

I think we could reshuffle things a little here by starting with your paragraph below on what turbo drive is in and then going into the good stuff you've written here about how it intercepts requests to only reload the parts it needs to. That answers the "What is the thing and why does it exist?" question that gets readers into the write frame.